Add java.lang.Iterable support for IN and NOT IN clauses in DSL

taichi opened the pull request:
This PR adds Iterable support for IN and NOT IN clauses in the Unified Criteria API's WHERE clause. 
For backward compatibility, existing List parameters have been kept as is.

Note: When using SQL files, Iterable parameters are already supported in Dao method arguments.

nakamura-to commented:
Thank you for your feedback. The `ComparisonDeclaration` class has overloaded versions of the `in` and `notIn` methods that accept tuples. Could you also apply the same fixes to these methods? Here are their signatures for reference:

- `<PROPERTY1, PROPERTY2> void in(Tuple2<PropertyMetamodel<PROPERTY1>, PropertyMetamodel<PROPERTY2>> left, List<Tuple2<PROPERTY1, PROPERTY2>> right)`
- `<PROPERTY1, PROPERTY2> void notIn(Tuple2<PropertyMetamodel<PROPERTY1>, PropertyMetamodel<PROPERTY2>> left, List<Tuple2<PROPERTY1, PROPERTY2>> right)`
- `<PROPERTY1, PROPERTY2, PROPERTY3> void in(Tuple3<PropertyMetamodel<PROPERTY1>, PropertyMetamodel<PROPERTY2>, PropertyMetamodel<PROPERTY3>> left, List<Tuple3<PROPERTY1, PROPERTY2, PROPERTY3>> right)`
- `<PROPERTY1, PROPERTY2, PROPERTY3> void notIn(Tuple3<PropertyMetamodel<PROPERTY1>, PropertyMetamodel<PROPERTY2>, PropertyMetamodel<PROPERTY3>> left, List<Tuple3<PROPERTY1, PROPERTY2, PROPERTY3>> right)`

Let me know if you need any additional details!

taichi commented:
I have added Iterable support for tuple-based IN and NOT IN methods.
I switched from TreeSet to LinkedHashSet in the tests since Tuple does not implement Comparable.

nakamura-to commented:
Thank you!
